services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java

```diff
@@ -134,6 +142,22 @@ public DefaultS3TransferManager(DefaultBuilder tmBuilder) {
         s3ClientType = s3CrtAsyncClient instanceof S3CrtAsyncClient ? S3ClientType.CRT_BASED : S3ClientType.JAVA_BASED;
     }
 
+    private static Supplier<S3AsyncClient> defaultS3AsyncClient() {
+        if (crtInClasspath()) {
+            return S3AsyncClient::crtCreate;
+        }
+        return S3AsyncClient::create;
+    }
+
+    private static boolean crtInClasspath() {
+        try {
+            ClassLoaderHelper.loadClass("software.amazon.awssdk.crt.s3.S3Client", false);
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+        return true;
+    }
+
```

Comment:
It's not very easy to add test for this logic because crt is always in classpath in this module. I'll do a manual one-off test for this

Reply:
>crt is always in classpath in this module
I did not understand this part, since its optional why will this be always in class path ?
Also can we add this in test modules like service test module etc ?

Also do we need to update developer guide or any other docs where we need to tell user to take crt dependency explicitly in pom.xml  ?

Reply:
I meant crt is always available in this module at compilation time, so it's not easy to add a test in this module (we'd need some way to exclude crt from classpath). We could add a test for this in a test module w/o adding crt dependency, but I'm not sure if it's worth the time and it seems flaky and fragile because in the future, we could add crt dependency in that module by accident and that will invalidate the test.

> Also do we need to update developer guide or any other docs where we need to tell user to take crt dependency explicitly in pom.xml ?

For dev guide, I already let Todd know and he will update his draft. Looks like I forgot about README. Will update it.